osu.Game/Screens/Select/DifficultyRecommender.cs

```diff
@@ -90,19 +88,22 @@ private void calculateRecommendedDifficulties()
         private IEnumerable<RulesetInfo> getBestRulesetOrder()
         {
             if (bestRulesetOrder != null)
-                return bestRulesetOrder;
+                return moveCurrentRulesetToFirst();
 
-            var otherRulesets = recommendedStarDifficulty.ToList()
-                                                         .Where(pair => !pair.Key.Equals(ruleset.Value))
-                                                         .OrderBy(pair => pair.Value)
-                                                         .Select(pair => pair.Key)
-                                                         .Reverse();
+            bestRulesetOrder = recommendedStarDifficulty.ToList()
+                                                        .OrderBy(pair => pair.Value)
+                                                        .Select(pair => pair.Key)
+                                                        .Reverse();
 
-            var rulesetList = new List<RulesetInfo>(new[] { ruleset.Value });
-            rulesetList.AddRange(otherRulesets);
+            return moveCurrentRulesetToFirst();
```

Comment:
Both return statements return the same thing, can make if statement into ``bestRulesetOrder == null``